src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            externalList.add(getEachInternalList(medicationDocument));
+        }
+        return externalList;
+    }
+
+    private Map<List<String>, Object> getEachInternalList
+            (QueryDocumentSnapshot medicationDocument) {
+        List<String> pks = new ArrayList<>();
+        String aux;
+        Map<List<String>, Object> internalList = new HashMap<>();
+        aux = medicationDocument.getId();
+        pks.add(pkToDate(aux));
+        pks.add(pkToName(aux));
+        internalList.put(Collections.singletonList(DATENAME), pks);
+        pks.clear();
+        internalList.put(Collections.singletonList(BODY),
+                medicationDocument.toObject(MedicationEntity.class));
+        return internalList;
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationsBetween(String owner, String petName,
+                                                                    String initialDate,
+                                                                    String finalDate)
+            throws DatabaseAccessException {
+        CollectionReference medsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        try {
+            getMeadicationsBetweenDatesFromDatabase(initialDate, finalDate, medsRef, externalList);
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        return externalList;
+    }
+
+    private void getMeadicationsBetweenDatesFromDatabase(String initialDate, String finalDate,
+                                                         CollectionReference medsRef,
+                                                         List<Map<List<String>, Object>> externalList)
+            throws InterruptedException, ExecutionException {
+        String date;
+        List<String> aux = null;
+        ApiFuture<QuerySnapshot> future = medsRef.get();
+        List<QueryDocumentSnapshot> medDocuments = future.get().getDocuments();
+        for (QueryDocumentSnapshot medDocument : medDocuments) {
+            aux = pkToList(medDocument.getId());
+            date = pkToDate(medDocument.getId());
+            if (initialDate.compareTo(date) < 0 && finalDate.compareTo(date) > 0) {
+                Map<List<String>, Object> internalList = new HashMap<>();
+                internalList.put(Collections.singletonList(DATENAME), aux);
+                internalList.put(Collections.singletonList(BODY),
+                        medDocument.toObject(MedicationEntity.class));
+                externalList.add(internalList);
+            }
+            aux.clear();
+        }
+    }
+
+
+
+    @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 24 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)